ledger/src/shred.rs

```diff
@@ -504,9 +504,10 @@ impl Shred {
     pub(crate) fn first_coding_index(&self) -> Option<u32> {
         match self.shred_type() {
             ShredType::Data => None,
-            // TODO should be: self.index() - self.coding_header.position
-            // once position field is populated.
-            ShredType::Code => Some(self.fec_set_index()),
+            ShredType::Code => {
+                let position = u32::from(self.coding_header.position);
```

Comment:
heh, I forget every time I've been away from this code for a while, but to double check:

There are two concepts, an index within the `nth` FEC set `FEC_(n,i)`, and an index `Sc_i` within the overall slot for coding shreds, and `Sd_i` for data shreds, where a slot is comprised of multiple FEC sets. Note here there is the nuance that the overall slot index `Sc_i` and `Sd_i` is indexed independently for coding and data shreds.

1. `self.coding_header.position` is the position of this shred within the coding set for this fec set, i.e. `FEC_(n,i)`
2. `self.index()` is the overall coding index of this shred within the entire slot, i.e. `S_i`

So `self.index().checked_sub(position)` is finding the coding shred index `Sc_i` of the first coding shred in this batch

In a follow-up could we add some documentation for this struct here: https://github.com/solana-labs/solana/blob/8db282f1a493fe50ea6b4510a86653acd79894b5/ledger/src/shred.rs#L221-L225 to explain what the fields represent in the updated protocol using more precise terminology like above, would really help for future reviews 😃 

For instance I think this might be outdated: https://github.com/solana-labs/solana/blob/8db282f1a493fe50ea6b4510a86653acd79894b5/ledger/src/shred.rs#L47

Reply:
Everything you mentioned sounds right to me except for the lat comment regarding the outdated code comment. Why is that comment outdated?

Reply:
Agreed with Behzad; I believe that we still need data / coding shreds to be same length:
https://github.com/rust-rse/reed-solomon-erasure/blob/2e134f6de13ca672cf185556c020ab997221b300/src/core.rs#L238-L239